src/main/java/uk/gov/hmcts/reform/iacasedocumentsapi/domain/handlers/presubmit/letter/InternalNonStandardDirectionLetterGenerator.java

```diff
@@ -0,0 +1,85 @@
+import uk.gov.hmcts.reform.iacasedocumentsapi.domain.entities.ccd.callback.PreSubmitCallbackResponse;
+import uk.gov.hmcts.reform.iacasedocumentsapi.domain.entities.ccd.callback.PreSubmitCallbackStage;
+import uk.gov.hmcts.reform.iacasedocumentsapi.domain.entities.ccd.field.Document;
+import uk.gov.hmcts.reform.iacasedocumentsapi.domain.handlers.PreSubmitCallbackHandler;
+import uk.gov.hmcts.reform.iacasedocumentsapi.domain.service.DocumentCreator;
+import uk.gov.hmcts.reform.iacasedocumentsapi.domain.service.DocumentHandler;
+import uk.gov.hmcts.reform.iacasedocumentsapi.infrastructure.DirectionFinder;
+
+@Component
+public class InternalNonStandardDirectionLetterGenerator implements PreSubmitCallbackHandler<AsylumCase> {
+    private final DocumentCreator<AsylumCase> internalNonStandardDirectionLetterCreator;
+    private final DocumentHandler documentHandler;
+    private final DirectionFinder directionFinder;
+
+    public  InternalNonStandardDirectionLetterGenerator(
+        @Qualifier("internalNonStandardDirectionLetter") DocumentCreator<AsylumCase> internalNonStandardDirectionLetterCreator,
+        DocumentHandler documentHandler,
+        DirectionFinder directionFinder) {
+        this.internalNonStandardDirectionLetterCreator = internalNonStandardDirectionLetterCreator;
+        this.documentHandler = documentHandler;
+        this.directionFinder = directionFinder;
+    }
+
+    public boolean canHandle(
+        PreSubmitCallbackStage callbackStage,
+        Callback<AsylumCase> callback
+    ) {
+        requireNonNull(callbackStage, "callbackStage must not be null");
+        requireNonNull(callback, "callback must not be null");
+
+        AsylumCase asylumCase = callback.getCaseDetails().getCaseData();
+
+        return callbackStage == PreSubmitCallbackStage.ABOUT_TO_SUBMIT
+            && callback.getEvent() == Event.SEND_DIRECTION
+            && isInternalCase(asylumCase)
+            && isAppellantInDetention(asylumCase)
+            && isRecipientAppellant(asylumCase);
+    }
+
+    public PreSubmitCallbackResponse<AsylumCase> handle(
+        PreSubmitCallbackStage callbackStage,
+        Callback<AsylumCase> callback
+    ) {
+        if (!canHandle(callbackStage, callback)) {
+            throw new IllegalStateException("Cannot handle callback");
+        }
+        final CaseDetails<AsylumCase> caseDetails = callback.getCaseDetails();
+        final AsylumCase asylumCase = caseDetails.getCaseData();
+
+        Document uploadDocument = internalNonStandardDirectionLetterCreator.create(caseDetails);
+
+        documentHandler.addWithMetadata(
+            asylumCase,
+            uploadDocument,
+            NOTIFICATION_ATTACHMENT_DOCUMENTS,
+            DocumentTag.INTERNAL_NON_STANDARD_DIRECTION_TO_APPELLANT_LETTER
+        );
+
+        return new PreSubmitCallbackResponse<>(asylumCase);
+    }
+
+    //method to check if the recipient is APPELLANT only
```

Comment:
dont really need this comment as its obvious from the method and name what it does.